src/chains/algorand/algorand.ts

```diff
@@ -0,0 +1,287 @@
+        const assetListSource = config.network.assetListSource;
+        Algorand._instances.set(
+          config.network.name,
+          new Algorand(
+            network,
+            nodeUrl,
+            indexerUrl,
+            assetListType,
+            assetListSource
+          )
+        );
+      } else {
+        throw new Error(
+          `Algorand.getInstance received an unexpected network: ${network}.`
+        );
+      }
+    }
+
+    return Algorand._instances.get(config.network.name) as Algorand;
+  }
+
+  public static getConnectedInstances(): { [name: string]: Algorand } {
+    const connectedInstances: { [name: string]: Algorand } = {};
+    if (this._instances !== undefined) {
+      const keys = Array.from(this._instances.keys());
+      for (const instance of keys) {
+        if (instance !== undefined) {
+          connectedInstances[instance] = this._instances.get(
+            instance
+          ) as Algorand;
+        }
+      }
+    }
+    return connectedInstances;
+  }
+
+  async getCurrentBlockNumber(): Promise<number> {
+    const status = await this._algod.status().do();
+    return status['next-version-round'];
+  }
+
+  public async getTransaction(txHash: string): Promise<PollResponse> {
+    const transactionId = txHash.startsWith('0x') ? txHash.slice(2) : txHash;
+    let currentBlock;
+    let transactionData;
+    let transactionBlock;
+    let fee;
+    try {
+      transactionData = await this._algod
+        .pendingTransactionInformation(transactionId)
+        .do();
+      transactionBlock = transactionData['confirmed-round']; // will be undefined if not confirmed
+      transactionBlock = transactionBlock ? transactionBlock : null;
+      fee = transactionData.txn.fee;
+      currentBlock = await this.getCurrentBlockNumber();
+    } catch (error: any) {
+      if (error.status != 404) {
+        throw error;
+      }
+      transactionData = await this._indexer
+        .lookupTransactionByID(transactionId)
+        .do();
+      currentBlock = transactionData['current-round'];
+      transactionBlock = transactionData.transaction['confirmed-round'];
+      fee = transactionData.transaction.fee;
+    }
+    return {
+      currentBlock,
+      txBlock: transactionBlock,
+      txHash: '0x' + transactionId,
+      fee,
+    };
+  }
+
+  public getAccountFromPrivateKey(mnemonic: string): Account {
+    return mnemonicToSecretKey(mnemonic);
+  }
+  async getAccountFromAddress(address: string): Promise<Account> {
+    const path = `${walletPath}/${this._chain}`;
+    const encryptedMnemonic: string = await fse.readFile(
+      `${path}/${address}.json`,
+      'utf8'
+    );
+    const passphrase = ConfigManagerCertPassphrase.readPassphrase();
+    if (!passphrase) {
+      throw new Error('missing passphrase');
+    }
+    const mnemonic = this.decrypt(encryptedMnemonic, passphrase);
+
+    return mnemonicToSecretKey(mnemonic);
+  }
```

Comment:
I modified those.
Existing implementation is 
```
public getAccountFromMnemonic(mnemonic: string): AlgorandAccount {
     const account = mnemonicToSecretKey(mnemonic);
     const address = account.addr;

     return {
       address,
       mnemonic,
     };
   }
   ```
   
   There's no point returning the mnemonic. Whatever function call it would have it already.
   It is better to return the objected returned by `mnemonicToSecretKey`

Reply:
Ok, understood